fiftyone/operators/loader.py

```diff
@@ -157,6 +162,15 @@ def exec_module_from_dir(module_dir, plugin_definition):
         )
     spec = importlib.util.spec_from_file_location(mod_dir, mod_filepath)
     mod = importlib.util.module_from_spec(spec)
+    try:
+        pctx = register_module(plugin_definition, mod)
+    except Exception as e:
+        raise e
+    if pctx.has_errors():
+        raise ValueError(
+            f"Plugin {plugin_definition.name} has errors: {pctx.errors}"
+        )
+    # only add to sys.modules if there are no errors
```

Comment:
This seems like the only changed needed - to only add to sys.modules if there are not errors, although I'm not sure you can attempt to load `exec_module` without it added, I've not tried that.